src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/RoutePattern/RoutePatternParser.cs

```diff
@@ -415,7 +417,7 @@ private RoutePatternLiteralNode ParseLiteral()
     {
         MoveBackBeforePreviousScan();
 
-        var literal = _lexer.TryScanLiteral();
+        var literal = _lexer.TryScanLiteral()!;
```

Comment:
Why can we be sure this is not null?

Reply:
The way parsing works in this specific area is that the char character has already been inspected. If it is a literal character, parsing rewinds and calls `TryScanLiteral`. A comment a few lines lower in this method explains why the result can't be null.